### DISCUSSION TITLE
Enabling organization wide environment variables

### DISCUSSION THREAD

**Discussion by GlatBane** [community/community#16259] [Enterprise] [answered]
I'm looking for a way as an Enterprise/Organization owner to be able to specify non-secret values globally so they are available to all workflows in the organization but not masked out in the log file. Essentially the same as secrets, but in clear text.

I would like to be able to specify a base set of environment variables.
Then I would like to specify overrides for specific environments
It should still be possible to specify an override in the workflow file

It is impractical to use secrets for this as they are masked out both in the UI and in logs, and we would like to be able to roll out updates across the Organization (unless there is an override in specific repo) with accepting the risk that things might break.

There exists a similar discussion, but that seems to be around a single repository. [#9249](https://github.com/github/feedback/discussions/9249#discussion-3763277), but I did not want to mix the repo and org split.

**Comment by HariSekhon** [comment]
+1

I also want this.

For example, right now I set `AWS_DEFAULT_REGION` as an organization-wide secret, even though an AWS region is really not a secret and should just be an environment variable, but it's currently the only way to have it uniformly set from one place across all repos since global environment variables don't exist.

In Jenkins, global environment variables are a basic feature for sharing common settings across lots of pipelines. It's basic DRY.

One problem with using organization-wide secrets in place of environment variables though, is that if constructing URLs from them to print, the masking will star parts of the URLs out, preventing users from getting the URLs they need in the output. It would require explicit unmasking, which is not cool, and there doesn't appear to be an unmask command the opposite of `"::add-mask::` command right now:

https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions#masking-a-value-in-log

**Comment by juliandunn** [comment, marked as answer]
We are working on this [right now](https://github.com/github/roadmap/issues/575). Thanks for all the feedback on this.
